From c4bf05f9d782da4b6e016797c3e3a52b8aabe381 Mon Sep 17 00:00:00 2001
From: Yu-An <yu-an.chen@quanta.corp-partner.google.com>
Date: Fri, 12 Sep 2025 08:49:25 +0800
Subject: [PATCH 03/19] matsu: Update fan and pwm configs

Update fan and pwm configs

BUG=b:443612246
TEST=zmake build matsu

Change-Id: Iae7a7e49433e3a2845ea85f1a9cb439f27ebb2de
---
 zephyr/program/ocelot/matsu/fan.dtsi        | 35 ++++++++++++++++++++
 zephyr/program/ocelot/matsu/generated.dtsi  | 36 +--------------------
 zephyr/program/ocelot/matsu/project.overlay | 19 ++---------
 zephyr/program/ocelot/matsu/pwm.dtsi        | 29 +++++++++++++++++
 4 files changed, 68 insertions(+), 51 deletions(-)
 create mode 100644 zephyr/program/ocelot/matsu/fan.dtsi
 create mode 100644 zephyr/program/ocelot/matsu/pwm.dtsi

diff --git a/zephyr/program/ocelot/matsu/fan.dtsi b/zephyr/program/ocelot/matsu/fan.dtsi
new file mode 100644
index 0000000000..53acef6f7a
--- /dev/null
+++ b/zephyr/program/ocelot/matsu/fan.dtsi
@@ -0,0 +1,35 @@
+/* Copyright 2025 The ChromiumOS Authors
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+/ {
+	fans {
+		compatible = "cros-ec,fans";
+
+		fan0: fan_0 {
+			pwms = <&pwm6 PWM_CHANNEL_6 PWM_KHZ(25) PWM_POLARITY_NORMAL>;
+			rpm-min = <2500>;
+			rpm-start = <2500>;
+			rpm-max = <6000>;
+			tach = <&tach0>;
+			rpm-deviation = <1>;
+			pgood-gpio = <&gpio_all_sys_pwrgd_r>;
+			enable-gpio = <&gpio_en_pp5000_fan>;
+		};
+	};
+};
+
+&pwm6 {
+	status = "okay";
+	prescaler-cx = <PWM_PRESCALER_C6>;
+	pinctrl-0 = <&pwm6_gpa6_default>;
+	pinctrl-names = "default";
+};
+
+&tach0 {
+	status = "okay";
+	pinctrl-0 = <&tach0a_gpd6_default>;
+	pinctrl-names = "default";
+	channel = <IT8XXX2_TACH_CHANNEL_A>;
+	pulses-per-round = <2>;
+};
diff --git a/zephyr/program/ocelot/matsu/generated.dtsi b/zephyr/program/ocelot/matsu/generated.dtsi
index a667daea99..302937176a 100644
--- a/zephyr/program/ocelot/matsu/generated.dtsi
+++ b/zephyr/program/ocelot/matsu/generated.dtsi
@@ -80,7 +80,7 @@
 			// signal name was manually overriden from "x00085"
 			gpios = <&gpioj 3 (GPIO_OUTPUT | GPIO_OPEN_DRAIN | GPIO_ACTIVE_HIGH | GPIO_OUTPUT_INIT_LOW)>;
 		};
-		gpio_ec_fan_pwr_disable_n_r: ec_fan_pwr_disable_n_r {
+		gpio_en_pp5000_fan: en_pp5000_fan {
 			gpios = <&gpioc 4 (GPIO_OUTPUT | GPIO_ACTIVE_HIGH | GPIO_OUTPUT_INIT_LOW)>;
 		};
 		gpio_ec_ish_int_odl_c_ec_ish_alert_r: ec_ish_int_odl_c_ec_ish_alert_r {
@@ -228,40 +228,6 @@
 	pinctrl-names = "default";
 };
 
-&pwm1 {
-	status = "okay";
-	prescaler-cx = <PWM_PRESCALER_C6>;
-	pinctrl-0 = <&pwm1_gpa1_default>;
-	pinctrl-names = "default";
-};
-
-&pwm2 {
-	status = "okay";
-	pinctrl-0 = <&pwm2_gpa2_default>;
-	pinctrl-names = "default";
-};
-
-&pwm3 {
-	status = "okay";
-	pinctrl-0 = <&pwm3_gpa3_default>;
-	pinctrl-names = "default";
-};
-
-&pwm5 {
-	status = "okay";
-	prescaler-cx = <PWM_PRESCALER_C7>;
-	pinctrl-0 = <&pwm5_gpa5_default>;
-	pinctrl-names = "default";
-};
-
-&tach0 {
-	status = "okay";
-	pinctrl-0 = <&tach0a_gpd6_default>;
-	pinctrl-names = "default";
-	channel = <IT8XXX2_TACH_CHANNEL_A>;
-	pulses-per-round = <2>;
-};
-
 &kbd {
 	status = "okay";
 	pinctrl-0 = <
diff --git a/zephyr/program/ocelot/matsu/project.overlay b/zephyr/program/ocelot/matsu/project.overlay
index 9c1b6b96aa..0098bfdc93 100644
--- a/zephyr/program/ocelot/matsu/project.overlay
+++ b/zephyr/program/ocelot/matsu/project.overlay
@@ -32,7 +32,7 @@ gpio_lid: &gpio_lid_open {};
 gpio_pwrbtn: &gpio_gsc_ec_pwr_btn_odl {};
 gpio_voldn: &gpio_ec_voldn_btn_odl {};
 gpio_volup: &gpio_ec_volup_btn_odl {};
-gpio_fan_control: &gpio_ec_fan_pwr_disable_n_r {};
+gpio_fan_control: &gpio_en_pp5000_fan {};
 gpio_all_sys_pwr_good: &gpio_all_sys_pwrgd_r {};
 gpio_temp_sensor_power_good: &gpio_en_s5_rails {};
 
@@ -44,6 +44,8 @@ cse_early_rec_sw: &gpio_c_mecc_cse_early_sw_r{};
 tach_fan0: &tach0 {};
 
 #include "adc.dtsi"
+#include "fan.dtsi"
+#include "pwm.dtsi"
 #include "../ite_hibernate.dtsi"
 #include "../rvp_usba.dtsi"
 #include "../rvp_battery.dtsi"
@@ -51,7 +53,6 @@ tach_fan0: &tach0 {};
 #include "../rvp_interrupts.dtsi"
 #include "../rvp_usbc.dtsi"
 #include "../rvp_keyboard.dtsi"
-#include "../rvp_fans.dtsi"
 #include "../rvp_temp_sensors.dtsi"
 #include "../rvp_ioex_gpio.dtsi"
 #include <ite/it8801-common-cfg.dtsi>
@@ -86,16 +87,6 @@ tach_fan0: &tach0 {};
 	status = "okay";
 };
 
-&pwm1 {
-	status = "okay";
-};
-&pwm2 {
-	status = "disabled";
-};
-&pwm3 {
-	status = "disabled";
-};
-
 /* Update board-specific references  */
 
 &pdc_rtk_c0 {
@@ -114,10 +105,6 @@ tach_fan0: &tach0 {};
 	irq-gpios = <&gpiof 5 GPIO_ACTIVE_LOW>;
 };
 
-&fan0 {
-	pwms = <&pwm5 PWM_CHANNEL_5 PWM_KHZ(25) PWM_POLARITY_NORMAL>;
-};
-
 &kblight {
 	pwms = <&pwm1 PWM_CHANNEL_1 PWM_HZ(324) PWM_POLARITY_NORMAL>;
 };
diff --git a/zephyr/program/ocelot/matsu/pwm.dtsi b/zephyr/program/ocelot/matsu/pwm.dtsi
new file mode 100644
index 0000000000..ff73848648
--- /dev/null
+++ b/zephyr/program/ocelot/matsu/pwm.dtsi
@@ -0,0 +1,29 @@
+/* Copyright 2025 The ChromiumOS Authors
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+/* kb back light */
+&pwm1 {
+	status = "okay";
+	prescaler-cx = <PWM_PRESCALER_C6>;
+	pinctrl-0 = <&pwm1_gpa1_default>;
+	pinctrl-names = "default";
+};
+
+/* gpio_led_1_l */
+&pwm2 {
+	status = "okay";
+	prescaler-cx = <PWM_PRESCALER_C6>;
+	pinctrl-0 = <&pwm2_gpa2_default>;
+	pinctrl-names = "default";
+};
+
+/* gpio_led_2_l */
+&pwm3 {
+	status = "okay";
+	prescaler-cx = <PWM_PRESCALER_C6>;
+	pinctrl-0 = <&pwm3_gpa3_default>;
+	pinctrl-names = "default";
+};
+
-- 
2.43.0

